app_runner.rb

```diff
@@ -56,4 +71,73 @@ def handle_list_rentals_for_person
     person_id = gets.chomp.to_i
     @app.list_rentals_for_person(person_id)
   end
+
+  def read_data_from_file(filename)
+    data = []
+    begin
+      file = File.open(filename, 'r')
+      data = JSON.parse(file.read)
+    rescue StandardError
+      file = File.open(filename, 'w')
+      file.write('[]')
+      file.close
+    end
+    data
+  end
+
+  def store_data(data, filename)
+    object_data = []
+    data.each do |o|
+      object_data << if o.is_a?(Rental)
+                       {
+                         book_title: o.book.title,
+                         person_id: o.person.id,
+                         date: Date.strptime(o.date, '%Y-%m-%d')
+                       }
+                     else
+                       o.to_object
+                     end
+    end
+
+    json_data = JSON.generate(object_data)
+    File.write(filename, json_data)
+  end
+
+  def load_people_from_file(filename)
+    people = read_data_from_file(filename)
+    people.each do |p|
+      if p['type'] == 'Teacher'
+        @app.people << Teacher.new(id: p['id'], name: p['name'], age: p['age'],
+                                   parent_permission: p['parent_permission'], specialization: p['specialization'])
+      else
+        @app.people << Student.new(id: p['id'], name: p['name'], age: p['age'],
+                                   parent_permission: p['parent_permission'], classroom: p['classroom'])
+      end
+    end
+  end
```

Comment:
- [ ] Good job so far 👏 . However, just as the previous reviewer stated, the code runs into an error when i tried to exit. Another new bug was that i could not run the code again after the exit. Kindly look into the action called upon when you quit the app and save your date in this file 👍 

### Error view:
![image](https://github.com/theresetuyi/OOP-school-library/assets/50990925/6dfa8f68-6422-4d8d-a380-bbc8f591be81)